packages/components/data-table/src/cell.styles.js

```diff
@@ -130,30 +133,49 @@ const BaseHeaderCell = styled.th`
   color: ${vars.colorSurface};
   background-color: ${vars.colorAccent};
 
-  position: ${(props) => (props.disableHeaderStickiness ? 'static' : 'sticky')};
+  position: ${(props) =>
+    props.disableHeaderStickiness ? 'relative' : 'sticky'};
   top: 0;
   z-index: 1;
 
   /* remove user-agent styles */
   padding: 0;
   font-weight: normal;
 
-  /* adds borders between header cells */
-  :not(:last-of-type) {
-    border-right: 1px solid ${vars.colorNeutral90};
-  }
+  /* right border that doesn't count towards the column width */
```

Comment:
Because otherwise, the `clientWidth` we get from measuring the columns will be one pixel smaller than the actual `width` it was supposed to have, because the border takes that space.

Visually, there's no difference between the border and this "fake border" box-shadow.